### PR TITLE
Replace `minimist-options` types with own ones

### LINKED ISSUE #134: Missing `array` in `flags.type` document
Hi,

Currently, the `flags.type` possible values are documented as `string`, `boolean`, and  `number`:

https://github.com/sindresorhus/meow/blob/5975fe618eb8dd70c06911d34ab52dd689a60616/index.d.ts#L11

Also on README:

https://github.com/sindresorhus/meow#flags

But the actual values by `minimist-options` include also `array`:

```ts
export type OptionType = 'string' | 'boolean' | 'number' | 'array';
```

https://github.com/vadimdemedes/minimist-options/blob/f6d24087f871d6605b6c8d0f9c2b36af18467875/index.d.ts#L3

Do these descriptions about `flags.type` need to include `array`?

### PR DESCRIPTION
**Pull request by ybiquitous**
Fix #134

See also:

https://github.com/vadimdemedes/minimist-options/blob/f6d24087f871d6605b6c8d0f9c2b36af18467875/index.d.ts

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -1,8 +1,21 @@
 import {PackageJson} from 'type-fest';
-import {Options as MinimistOptions} from 'minimist-options';
 
 declare namespace meow {
-	interface Options<Flags extends MinimistOptions> {
+	type FlagType = 'string' | 'boolean' | 'number';
+
+	interface Flag<Type extends FlagType, Default> {
+		readonly type?: Type;
+		readonly alias?: string;
```
Comment (ybiquitous) [inline index.d.ts]: Considering #132, we may change like this:

```suggestion
		readonly alias?: string | ReadonlyArray<string>;
```

File: index.d.ts [hunk 2]
```diff
@@ -1,8 +1,21 @@
...
+		readonly default?: Default;
+	}
+
+	type StringFlag = Flag<'string', string>;
+	type BooleanFlag = Flag<'boolean', boolean>;
+	type NumberFlag = Flag<'number', number>;
+
+	type BaseFlags = { [key: string]: StringFlag | BooleanFlag | NumberFlag };
```
Comment (ybiquitous) [inline index.d.ts]: I don't really like the name `BaseFlags`. Do you have another good name?

Comment (sindresorhus) [reply index.d.ts]: `AnyFlags`?

File: index.d.ts (hunk 1 above)
Comment (sindresorhus) [reply index.d.ts]: That should be done in that PR though.

File: index.d.ts (hunk 2 above)
Comment (ybiquitous) [reply index.d.ts]: Looks nice! Thank you, I'll change.

File: index.d.ts (hunk 1 above)
Comment (ybiquitous) [reply index.d.ts]: Agreed. 👍

**Comment by sindresorhus** [comment]
Thanks :)

**Comment by ybiquitous** [comment]
You are welcome!

**Pull request merged**